docs/pair_hmm.tex

```diff
@@ -0,0 +1,101 @@
+\begin{equation}
+P(\mc{R} | \mc{A}, \mc{H}) = \prod_k P(r_{i_k} | h_{j_k}, q_{i_k})^{{\rm I}[s_k = M]},
+\end{equation}
+where $r_m$ and $h_n$ are the $m$th read bases and $n$th haplotype base and $q_m$ is the quality of the read's $m$th base.  Note that this only includes alignments in the match state.  The per-base emission is given directly from the definition of base quality:
+\begin{equation}
+P(b_2 | b_1, q) = \left\{ \begin{array}{cc} \epsilon(q)/3 & (b_1 \ne b_2) \\ 1 - \epsilon(q) & (b_1 = b_2)  \end{array} \right.
+\end{equation}
+
+\section{Dynamic Programming}
+
+Define the matrices $M$, $I$ and $D$ by $M_{ij} = $ the total likelihood of \textit{all} paths from the beginning of the read to position $i$ that end in a match state, and likewise for $D$ and $I$.  Then the recursions
+\begin{align}
+M_{ij} =& P(r_i | h_j, q_i) \left( M_{i-1,j-1} T_{MM} + I_{i-1,j-1} T_{IM} + D_{i-1,j-1} T_{DM} \right) \label{match} \\
+I_{ij} =& M_{i-1,j} T_{MI} + I_{i-1,j} T_{II} \label{ins} \\
+D_{ij} =& M_{i,j-1} T_{MD} + D_{i, j - 1} T_{DD} \label{del}
+\end{align}
+define the entire pair HMM algorithm:
+
+\begin{algorithm}
+\begin{algorithmic}[1]
+\State Initialize $M_{0,j} = I_{0,j} = 0$ and $D_{0,j} = 2^{1020}/|\mc{H}|$ for $1 \le j \le |\mc{H}|$.
+\For{$1 \le i \le | \mc{R} |$}
+	\For{$1 \le j \le | \mc{H} |$}
+		\State Calculate $M_{ij}$ via Equation \ref{match}.
+		\State Calculate $I_{ij}$ via Equation \ref{ins}.
+		\State Calculate $D_{ij}$ via Equation \ref{del}.
+	\EndFor
+\EndFor
+\State Total likelihood $P(\mc{R} | \mc{H})$ is $\sum_j \left(M_{\mc{R},j} + I_{\mc{R},j} \right)$.
+\end{algorithmic}
+\caption{Pair HMM algorithm}
+\label{pairHMM}
+\end{algorithm}
+
+That the $i = 0$ rows of $M$ and $I$ are initialized to zero corresponds to starting at an imaginary position one base before the read start in a deletion state.  The factor of $1/|\mc{H}|$ corresponds to a flat prior on which $j$ an alignment starts at.  The curious factor of $2^{1020}$ is a huge number to prevent underflow -- all multiplications are by numbers less than 1, so we needn't worry about overflow -- which is a much more efficient approach than performing the computation in log space.  The omission of $D$ from the returned value recognizes the fact that a terminal deletion is meaningless.
```

Comment:
I would phrase this differently by pointing out that those boundary conditions are set that way because this is a local alignment and we don't want a penalty associated with where the read starts with respect to the beginning of the haplotype.